modules/modify-data-object/pkg/dataobject/provider.go

```diff
@@ -155,6 +170,20 @@ func (d *DataObjectCreator) CreateDataObject() (*unstructured.Unstructured, erro
 func (d *DataObjectCreator) waitForSuccessDv(namespace, name string) error {
 	return wait.PollImmediate(constants.PollInterval, constants.PollTimeout, func() (bool, error) {
 		dv, err := d.dataObjectProvider.GetDv(namespace, name)
+
+		if errors.ReasonForError(err) == metav1.StatusReasonNotFound {
+			pvc, err := d.dataObjectProvider.GetPVC(namespace, name)
+			if err != nil {
+				return false, err
+			}
+
+			if pvc.Status.Phase == v1.ClaimPending || pvc.Status.Phase == v1.ClaimBound {
```

Comment:
No way `pvc.Status.Phase == v1.ClaimPending` if DV was alreday GC'ed.
If DV was not found I would simply check PVC exists, that's all.

Reply:
It's ok on most cases (except smart clone where pvc doesn't have this annotation)

Reply:
So as you said, the most robust solution will be to check for the PVC to exist if no DV was not found?

Reply:
right, this is more than enough